scripts/migrations/2020-08-update_migration_created_date.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+import json
+from botocore.exceptions import ClientError
+
+from common import get_aws_resource, scan_table
+
+READ_ONLY_ROLE_ARN = "arn:aws:iam::975596993436:role/storage-read_only"
+dynamodb = get_aws_resource("dynamodb", role_arn=READ_ONLY_ROLE_ARN)
+s3 = get_aws_resource("s3", role_arn=READ_ONLY_ROLE_ARN)
+
+vhs_table = "vhs-storage-staging-manifests-2020-07-24"
+
+backfill_vhs_table = "vhs-storage-staging-manifests-2020-08-19"
+
+
+def get_bucket_key(item):
+    try:
+        bucket = item["payload"]["namespace"]
+        key = item["payload"]["path"]
+    except KeyError:
+        try:
+            bucket = item["payload"]["bucket"]
+            key = item["payload"]["key"]
+        except:
+            raise RuntimeError(f"Find s3 bucket, key in {item}")
+        else:
+            return bucket, key
+    else:
+        return bucket, key
+
+
+def get_vhs_json(bucket, key):
+    return json.loads(s3.Object(bucket, key).get()["Body"].read().decode("utf-8"))
+
+
+def put_vhs_json(bucket, key, content):
+    try:
+        s3object = s3.Object(bucket, key)
+        s3object.put(Body=(bytes(json.dumps(content).encode("UTF-8"))))
+    except ClientError as e:
+        raise RuntimeError(f"Error updating backfill vhs object {bucket} {key}: {e}")
+
+
+for item in scan_table(TableName=vhs_table):
+    id = item["id"]
+    version = item["version"]
+    bucket, key = get_bucket_key(item)
+    try:
+        vhs_content = get_vhs_json(bucket, key)
+    except:
+        raise RuntimeError(f"Cannot read s3 entry for {id}: {item}")
+    else:
```

Comment:
It might be worth a bit here that checks for three replicas in the manifest already in VHS, and if they're there, it skips copying the createdDate across. That way, you can step through it, one bag at a time if necessary.

That’s typically how I write this sort of script – I write it so that it skips redoing work that’s already been done, then I use `break` to do a single step at a time until I'm happy it works.

Reply:
makes sense, thanks

Reply:
Actually, I'm not sure this is true. We have sent bags with Azure replication enabled but without the backfill infra in place. So there might be bags for which the current VHS has 3 replicas but the createdDate in the backfill one is still incorrect. I think it would be better to leave it doing the work more than once and not optimize in this instance. if the db is too big in prod we can always parallelize right?